### PR TITLE
Translation tab

### PR DESCRIPTION
**Pull request by miri**
#60

### CODE REVIEW

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 1]
```diff
@@ -16,6 +16,59 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
...
+    }
+    $author_names .= ', '.$author_wrapper->field_german_name->value();
+  }
+  if ($author_wrapper->field_pseudonym) {
+    if (!isset($author_names_as_field)) {
+      $author_names_as_field = field_view_field('node', $author_node, 'field_pseudonym');
+    }
+    $author_names .= ', '.$author_wrapper->field_pseudonym->value();
+  }
+  // If Success build field, change value to all fields values together.
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: Please create an array of names, and then transform it to a string with commas using ```implode()```

**Comment by bitamar** [comment]
Merged partially - Without the changes to haskala_book.module

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 2]
```diff
@@ -16,6 +16,56 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
...
 
 /**
+ *  Translation tab - Other names for the original author.
+ *
+ *  @param $person_nid
+ *   The author person node ID.
+ *
+ *  @return  The created field for orginal author other names.
+ */
+function haskala_book_author_names($person_nid) {
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: This function should probably be named haskala_person_get_name() or something similar, and be part of haskala_person

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 3]
```diff
@@ -16,6 +16,56 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
...
+  $query = new EntityFieldQuery();
+  $query->entityCondition('entity_type', 'node')
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: Is the query for loading a single node? Use``` $wrapper = entity_metadata_wrapper('node', $person_nid);``` then.

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 4]
```diff
@@ -16,6 +16,56 @@ function haskala_book_ctools_plugin_directory($module, $plugin) {
...
+  $author_names = array();
+
+  // Try build field_view_field().
+  if ($author_wrapper->field_hebrew_name) {
+    $author_names_as_field = field_view_field('node', $author_node, 'field_hebrew_name');
+    $author_names[] = $author_wrapper->field_hebrew_name->value();
+  }
+  if ($author_wrapper->field_german_name) {
+    if (!isset($author_names_as_field)) {
+      $author_names_as_field = field_view_field('node', $author_node, 'field_german_name');
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: Probably we don't need to use field_view_field three times. Better would be to return the name as a string, and handle the label elsewhere

File: haskala/modules/custom/haskala_book/haskala_book.module [hunk 5]
```diff
@@ -81,6 +131,11 @@ function haskala_book_get_tab_fields($wrapper, $tab, $node_view, $book_nid) {
       $field = render($node_view[$field_name]);
       $tab_fields[] = $field;
 
+      // Other names for the author
```
Comment (bitamar) [inline haskala/modules/custom/haskala_book/haskala_book.module]: Add dot

File: haskala/modules/custom/haskala_book/haskala_book.module (hunk 4 above)
Comment (miri) [inline haskala/modules/custom/haskala_book/haskala_book.module]: There is checking of  !isset(), So only one time it use  field_view_field.

File: haskala/modules/custom/haskala_people/haskala_people.module [hunk 6]
```diff
@@ -5,3 +5,43 @@
...
+  $person_names = array();
+
+  // Try build field_view_field().
+  if ($person_wrapper->field_hebrew_name) {
+    $person_names_as_field = field_view_field('node', $person_node, 'field_hebrew_name');
+    $person_names[] = $person_wrapper->field_hebrew_name->value();
+  }
+  if ($person_wrapper->field_german_name) {
+    if (!isset($person_names_as_field)) {
+      $person_names_as_field = field_view_field('node', $person_node, 'field_german_name');
```
Comment (bitamar) [inline haskala/modules/custom/haskala_people/haskala_people.module]: Probably we don't need the three calls to field_view_field. Instead, build the name as a string, and convert it into an array once

**Comment by bitamar** [comment]
@miri - Please merge squash

**Comment by miri** [comment]
Merged.

**Pull request closed without merging**